src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -3994,6 +4050,235 @@ module Mutations = struct
     ; archive_extensional_block
     ; send_rosetta_transaction
     ]
+
+  module Itn = struct
+    (* ITN-specific mutations *)
+
+    let scheduler_tbl : unit Async_kernel.Ivar.t Uuid.Table.t =
+      Uuid.Table.create ()
+
+    let schedule_payments =
+      io_field "schedulePayments"
+        ~args:
+          Arg.
+            [ arg "input" ~doc:"Payments details"
+                ~typ:(non_null Types.Input.Itn.PaymentDetails.arg_typ)
+            ]
+        ~typ:(non_null string)
+        ~resolve:(fun { ctx = mina; _ } () input ->
+          return
+          @@
+          match input with
+          | Error err ->
+              Error (sprintf "Invalid input to payment scheduler: %s" err)
+          | Ok payment_details ->
+              let max_memo_len = Signed_command_memo.max_input_length in
+              if List.is_empty payment_details.senders then
+                Error "Empty list of senders"
+              else if String.length payment_details.memo > max_memo_len then
+                Error
+                  (sprintf "Memo too long, limited to %d characters"
+                     max_memo_len )
+              else if
+                Currency.Fee.( < ) payment_details.fee_max
+                  payment_details.fee_min
+              then Error "Maximum fee less than mininum fee"
+              else
+                let logger = Mina_lib.top_level_logger mina in
+                let senders = payment_details.senders |> Array.of_list in
+                let num_senders = Array.length senders in
+                let sources =
+                  Array.map senders ~f:(fun sender ->
+                      Signature_lib.Public_key.of_private_key_exn sender
+                      |> Signature_lib.Public_key.compress )
+                in
+                Option.value_map (get_ledger_and_breadcrumb mina)
+                  ~default:(Error "Could not get best tip ledger")
+                  ~f:(fun (ledger, _tip) ->
+                    let nonce_opts =
+                      Array.map sources ~f:(fun source ->
+                          let open Option.Let_syntax in
+                          let acct_id =
+                            Account_id.create source Token_id.default
+                          in
+                          let%bind loc =
+                            Ledger.location_of_account ledger acct_id
+                          in
+                          let%map { nonce; _ } = Ledger.get ledger loc in
+                          nonce )
+                      |> Array.zip_exn sources
+                    in
+                    let missing_nonces =
+                      Array.filter nonce_opts ~f:(fun (_source, nonce_opt) ->
```

Comment:
(Nit) Use `filter_map` and remove both `List.map` usages in `missing_nonce_pks` ?